frontend/js/src/explore/artist-similarity/Data.tsx

```diff
@@ -0,0 +1,244 @@
+  target: string;
+  distance: number;
+};
+
+type GraphDataType = {
+  nodes: Array<NodeType>;
+  links: Array<LinkType>;
+};
+
+const colorGenerator = (): tinycolor.Instance => {
+  return tinycolor(`hsv(${Math.random() * 360}, 90%, 90%)`);
+};
+
+function Data() {
+  const ARTIST_MBID = "8f6bd1e4-fbe1-4f50-aa9b-94c450ec0f11";
+  const SIMILAR_ARTISTS_LIMIT_VALUE = 18;
+  const BASE_URL =
+    "https://labs.api.listenbrainz.org/similar-artists/json?algorithm=session_based_days_7500_session_300_contribution_5_threshold_10_limit_100_filter_True_skip_30&artist_mbid=";
+  // Serves as the maximum distance between nodes
+  const LINK_DIST_MULTIPLIER = 250;
+  // Serves as the minimum distance between nodes
+  const MIN_LINK_DIST = 0;
+  // Size of the main node
+  const MAIN_NODE_SIZE = 150;
+  // Size of the similar nodes
+  const SIMILAR_NODE_SIZE = 85;
+  // Apha value of the background color of the graph
+  const BACKGROUND_ALPHA = 0.2;
+  const COLOR_MIX_WEIGHT = 0.3;
+  // Score in case it is undefined (as in case of main artist)
+  const NULL_SCORE = Infinity;
+
+  const color1 = colorGenerator();
+  const color2 = color1.clone().tetrad()[1];
+  const [similarArtistsList, setSimilarArtistsList] = useState<
+    Array<ArtistType>
+  >([]);
+  // State to store the complete list of similar artists to help with when user changes artist limit
+  const [completeSimilarArtistsList, setCompleteSimilarArtistsList] = useState<
+    Array<ArtistType>
+  >([]);
+  const [mainArtist, setMainArtist] = useState<ArtistType>();
+  const [similarArtistsLimit, setSimilarArtistsLimit] = useState(
+    SIMILAR_ARTISTS_LIMIT_VALUE
+  );
+  const [colors, setColors] = useState([color1, color2]);
+  const [artistMBID, setArtistMBID] = useState(ARTIST_MBID);
+
+  const scoreList: Array<number> = [];
+
+  const processData = useCallback(
+    (dataResponse: ApiResponseType): void => {
+      // Type guard for dataset response
+      const isDatasetResponse = (
+        response: MarkupResponseType | DatasetResponseType
+      ): response is DatasetResponseType => {
+        return response.type === "dataset";
+      };
+      // Get the datasets out of the API response
+      const artistsData = dataResponse.filter(isDatasetResponse);
+      if (artistsData.length) {
+        // Get the main artist from the first dataset
+        setMainArtist(artistsData[0].data[0]);
+        // Get the similar artists from the second dataset
+        const similarArtistsResponse = artistsData[1];
+        if (similarArtistsResponse?.data?.length) {
+          setCompleteSimilarArtistsList(similarArtistsResponse.data);
+          setSimilarArtistsList(
+            similarArtistsResponse.data.slice(0, similarArtistsLimit)
+          );
+        }
+        // In case no similar artists are found
+        else {
+          setSimilarArtistsList([]);
+        }
+      }
+      setColors([tinycolor.mix(color1, color2, COLOR_MIX_WEIGHT), color2]);
+    },
+    [artistMBID]
```

Comment:
The dependency list needs to be updated.

Reply:
Fixed